### PR TITLE
Bug 1834343: router: Re-enable h2spec and grpc-interop e2e tests

### PR DESCRIPTION
**Pull request by frobware**
This re-enables the h2spec and grpc-interop tests for the router. They were disabled by PR #24913. The way http/2 support is enabled was changed in https://github.com/openshift/router/pull/125 and this PR reacts to those changes.

### CODE REVIEW

**Comment by frobware** [comment]
/hold

I notice testing this elsewhere today that the grpc tests are failing. Not clear why they work in the PR.

**Comment by frobware** [comment]
/hold cancel

Latest commits adds a readiness probe to grpc-interop and h2spec pods.

**Comment by frobware** [comment]
@Miciah the changes here are the bare minimum based on our discussion last week; certs added, readiness probes added. I think we should collapse h2spec to just re-encrypt as a separate PR.

File: test/extended/testdata/router/router-grpc-interop.yaml [hunk 1]
```diff
@@ -134,11 +137,22 @@ objects:
...
         value: "info"
+      readinessProbe:
+        httpGet:
+          path: /healthz
+          port: 8080
+        initialDelaySeconds: 10
+        periodSeconds: 3
+      env:
+      - name: GODEBUG
+        value: http2debug=1
```
Comment (Miciah) [inline test/extended/testdata/router/router-grpc-interop.yaml:148]: You define `env` twice; I think this definition needs to be merged into the one above:  https://github.com/openshift/origin/blob/d9b8b7fa56c3a4d36584d6055de5a2d602b37d3e/test/extended/testdata/router/router-grpc-interop.yaml#L135-L139

Comment (frobware) [reply test/extended/testdata/router/router-grpc-interop.yaml]: Thanks. Done.

**Comment by frobware** [comment]
```
Installing from release registry.svc.ci.openshift.org/ci-op-r6ct7yzq/release@sha256:b26f7ef255831df65df2736a35f8f890752db08c85a98060ea99fdffec21322b
level=info msg="Credentials loaded from environment variable \"GOOGLE_CLOUD_KEYFILE_JSON\", file \"/etc/openshift-installer/gce.json\""
level=info msg="Consuming Install Config from target directory"
level=warning msg="Found override for release image. Please be warned, this is not advised"
level=info msg="Credentials loaded from environment variable \"GOOGLE_CLOUD_KEYFILE_JSON\", file \"/etc/openshift-installer/gce.json\""
level=fatal msg="failed to fetch Terraform Variables: failed to generate asset \"Terraform Variables\": failed to get GCP public zone: failed to list DNS Zones: googleapi: Error 503: Backend Error, backendError"
```
/retest

**Comment by frobware** [comment]
```
Failing tests:

[sig-cli][Feature:LegacyCommandTests][Disruptive][Serial] test-cmd: test/cmd/projects.sh [Suite:openshift]

2020/05/20 15:36:07 Container test in pod e2e-cmd failed, exit code 1, reason Error
2020/05/20 15:46:45 Copied 147.90MB of artifacts from e2e-cmd to /logs/artifacts/e2e-cmd
2020/05/20 15:46:45 Releasing lease for "gcp-quota-slice"
2020/05/20 15:46:45 No custom metadata found and prow metadata already exists. Not updating the metadata.
2020/05/20 15:46:46 Ran for 1h41m36s
error: some steps failed:
  * could not run steps: step e2e-cmd failed: template pod "e2e-cmd" failed: the pod ci-op-qnsmrd7f/e2e-cmd failed after 1h12m15s (failed containers: test): ContainerFailed one or more containers exited

Container test exited with code 1, reason Error
---
```
/retest

**Pull request merged**